### DISCUSSION TITLE
MikroORM config file not found in ['./dist/mikro-orm.config.js', './mikro-orm.config.js']

### DISCUSSION THREAD

**Discussion by kulame** [mikro-orm/mikro-orm#3337] [Q&A]
**Describe the bug**

this is my repo  [https://github.com/kulame/vpnkit](nestjs api)

i followed the document https://mikro-orm.io/docs/installation#setting-up-the-commandline-tool

this is my mikro-orm config

```typescript
  "mikro-orm": {
    "useTsNode": true,
    "configPaths": [
      "./libs/database/src/mikro-orm.config.ts",
      "./dist/mikro-orm.config.js"
    ]
  },
```
when i run it

```bash
nest start api
```

it show some erorrs 

**Stack trace**
```bash
[Nest] 47308  - 2022/07/27 19:12:33   ERROR [ExceptionHandler] MikroORM config file not found in ['./dist/mikro-orm.config.js', './mikro-orm.config.js']
Error: MikroORM config file not found in ['./dist/mikro-orm.config.js', './mikro-orm.config.js']
```

**To Reproduce**
1. git clone git@github.com:kulame/vpnkit.git
2. cd vpnkit && pnpm install
3. nest start api

**Expected behavior**
auto generate the ./dist/mikro-orm.config.js file

| Dependency | Version |
| - | - |
| node | v18.4.0 |
| typescript | v4.7.4 |
| mikro-orm | 5.2.4 |

**Comment by B4nan** [comment]
> auto generate the ./dist/mikro-orm.config.js file

That's not how it works, you are the one who needs to compile your project, don't expect ORM to do it for you (why should it? its a database tool...).

**Reply by kulame** [reply]
when i build it by nest build api,
it generate nothing.

and i use tcs to compile the ts file, move it to the dist dir.
it show error 
```bash
[Nest] 58022  - 2022/07/27 20:00:53   ERROR [ExceptionHandler] Source file './dist/entities/BatchHistory.ts' not found. Check your 'entitiesTs' option and verify you have 'compilerOptions.declaration' enabled in your 'tsconfig.json'. If you are using webpack, see https://bit.ly/35pPDNn
```

but i have the file in dist dir
```bash
➜  dist git:(main) ✗  find * | grep entities/BatchHistory.ts 
entities/BatchHistory.ts
```

**Reply by B4nan** [reply]
> when i build it by nest build api, it generate nothing.

Well, thats not really _a bug in the ORM_, right? You probably have TS misconfigured or the conifig file is in a place where its not considered for compitation.

> and i use tcs to compile the ts file, move it to the dist dir. it show error

You cant compile one file and move it and expect things to work. The paths are obviously wrong if you changed the file location.

Also the fact that a compiled config is trying to load a TS entity file points to other errors you have in the config or generally in your setup. 

I also dont understand why you expect a file from `libs/database/src/mikro-orm.config.ts` to be compiled to just `dist/mikro-orm.config.js`.

**Reply by B4nan** [reply]
Maybe try to look at some example projects, where things are properly set up. First run them to see they work, then start slowly modifying them and iterate on that. This looks like you tried to set up many things at once without testing.

https://github.com/mikro-orm/nestjs-realworld-example-app
